### PR TITLE
Fix extractjar task  ci 

### LINKED ISSUE #33201: Execution failed for task ':test:fixtures:krb5kdc-fixture:extractJar'
This one failed four times in the last ten days. Last two failures are: 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=centos/2628/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=centos/2637/console

Looks like a test infra problem:

```
02:53:54 java.nio.file.NoSuchFileException: /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob-unix-compatibility/os/centos/test/fixtures/krb5kdc-fixture/build/distributions/krb5kdc-fixture-7.0.0-alpha1-SNAPSHOT.jar
02:53:54 	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
02:53:54 :test:fixtures:krb5kdc-fixture:extractJar (Thread[Task worker for ':' Thread 6,5,main]) completed. Took 0.095 secs.
02:53:54 3917 actionable tasks: 2565 executed, 1352 up-to-date
02:53:54 	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
02:53:54 	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
02:53:54 	at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
02:53:54 	at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:145)
02:53:54 	at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
02:53:54 	at java.base/java.nio.file.Files.readAttributes(Files.java:1755)
02:53:54 	at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1220)
02:53:54 	at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:727)
02:53:54 	at java.base/java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:845)
02:53:54 	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:245)
02:53:54 	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:175)
02:53:54 	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:146)
02:53:54 	at jdk.jartool/sun.tools.jar.Main.extract(Main.java:1377)
02:53:54 	at jdk.jartool/sun.tools.jar.Main.run(Main.java:396)
02:53:54 	at jdk.jartool/sun.tools.jar.Main.main(Main.java:1669)
```

### PR DESCRIPTION
**Pull request by alpar-t**
I don't see anything wrong with how the task is being set up.
I suspect there's a race condition with Gradle's lazy task configuration as we have seen before, 
but instead spending all the time tracking it down, it makes more sense to move this to an integration test instead. 

Closes #33201

### CODE REVIEW

File: buildSrc/src/test/java/org/elasticsearch/gradle/BuildPluginIT.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        assertBuildFileExists(result, "elasticsearch.build", "distributions/elasticsearch.build.jar");
+
+        try (ZipFile zipFile = new ZipFile(new File(
+            getBuildDir("elasticsearch.build"), "distributions/elasticsearch.build.jar"
+        ))) {
+            ZipEntry licenseEntry = zipFile.getEntry("META-INF/LICENSE.txt");
+            ZipEntry noticeEntry = zipFile.getEntry("META-INF/NOTICE.txt");
+            assertNotNull("Jar does not have META-INF/LICENSE.txt", licenseEntry);
+            assertNotNull("Jar does not have META-INF/NOTICE", noticeEntry);
```
Comment (rjernst) [inline buildSrc/src/test/java/org/elasticsearch/gradle/BuildPluginIT.java]: NOTICE -> NOTICE.txt

File: buildSrc/src/test/java/org/elasticsearch/gradle/BuildPluginIT.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            try (
+                InputStream licese = zipFile.getInputStream(licenseEntry);
```
Comment (rjernst) [inline buildSrc/src/test/java/org/elasticsearch/gradle/BuildPluginIT.java]: typo: licese -> license

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy [hunk 3]
```diff
@@ -25,20 +25,8 @@ import org.apache.tools.ant.taskdefs.condition.Os
...
-import org.gradle.api.Task
-import org.gradle.api.XmlProvider
-import org.gradle.api.artifacts.Configuration
-import org.gradle.api.artifacts.Dependency
-import org.gradle.api.artifacts.ModuleDependency
-import org.gradle.api.artifacts.ModuleVersionIdentifier
-import org.gradle.api.artifacts.ProjectDependency
-import org.gradle.api.artifacts.ResolvedArtifact
-import org.gradle.api.artifacts.SelfResolvingDependency
+import org.gradle.api.*
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: Please revert converting to wildcard imports...

**Review by rjernst — APPROVED** [review]
LGTM, just one main thing

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy (hunk 3 above)
Comment (atorok) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy]: Groovy has a different setting for this, I was under the impression that I already had it configured,
but looks like it was the case only for java.

**Comment by atorok** [comment]
@elasticmachine test thsi pelase

**Comment by atorok** [comment]
@elasticmachine test this please

**Comment by atorok** [comment]
@elasticmachine test this please

**Pull request merged**